pandas/core/frame.py

```diff
@@ -3265,6 +3265,19 @@ def _get_info_slice(obj, indexer):
         for dtypes in (include, exclude):
             invalidate_string_dtypes(dtypes)
 
+        def add_extension_types(dtypes, search_dtype, func):
+            """Adds bool or numeric extension types to include/exclude"""
+            extension_dtypes = [dtype.type for dtype in self.dtypes
+                                if is_extension_array_dtype(dtype) and
+                                func(dtype)]
+            if search_dtype in dtypes:
+                return frozenset(dtypes.union(extension_dtypes))
+            else:
+                return dtypes
+        include = add_extension_types(include, np.number, is_numeric_dtype)
```

Comment:
I don't quite understand this. Are you unconditionally adding adding an EA if it's considered numeric here?